### DISCUSSION TITLE
Evaluator CRUD in UI and API

### DISCUSSION THREAD

**Discussion by qrosseel** [langfuse/langfuse#4096] [Ideas]
### Describe the feature or potential improvement

I'm not sure if I'm missing something, but I cannot seem to do CRUD actions on `Evaluator` objects, either through the UI or API. 
Is there active work in this area? 

**Problem**
- Changing the query & generation parameters requires me to create a new Evaluator from a template all together, which is a frustrating UX. When I need to change a configuration (e.g. on what traces it is triggered), I'm not able to edit all params of the evaluator. 
- Deleting an `Evaluator` just turns them `Inactive`, leaving a trace of messy development evaluators that didn't work
- AFAIK, there's no documentation on a public API for CRUD on these objects 

![image](https://github.com/user-attachments/assets/789b6ffa-299e-49ba-96da-0d50af92aac5)

**Comment by marliessophie** [comment]
Hey @qrosseel, thanks for reporting. 

We're adding support for updates over the next weeks. I will ping here once released. We're planning to support update operations, but are currently not planning to support deleting. Appreciate the feedback though, have logged an internal ticket and will keep you posted here

**Reply by ggdupont** [reply]
@marliessophie will that also cover accessing the evaluator from the API? and python SDK?

**Reply by ohayolink** [reply]
@marliessophie  When will the CRUD operations on evaluator be implemented?

**Reply by marliessophie** [reply]
@davidmc971 is taking this over on our side, will let him provide timeline update :)

**Comment by nishantkr18** [comment]
Another request for CRUD operations on evaluator. :hand:
